### PR TITLE
Change relation of meeting with user table

### PR DESCRIPTION
**Pull request by sim4858**
Change Participant to User
Add Expense class
Change relation of meeting with user table , Many to Many
Data Base test done

### CODE REVIEW

File: src/main/java/us/controller/MeetingController.java [hunk 1]
```diff
@@ -7,11 +7,13 @@
...
 import us.model.Meeting;
-import us.model.Participant;
+import us.model.User;
+import us.repository.ExpenseRepository;
 import us.repository.MeetingRepository;
-import us.repository.ParticipantRepository;
+import us.repository.UserRepository;
 
 import javax.servlet.http.HttpSession;
+import javax.transaction.Transactional;
```
Comment (NESOY) [inline src/main/java/us/controller/MeetingController.java]: Plz Remove Unnecessary Import Code

File: src/main/java/us/controller/MeetingController.java [hunk 2]
```diff
@@ -34,10 +38,9 @@ public String createMeeting(Meeting meeting,HttpSession session) {
         meetingRepository.save(meeting);
         String base_url = "http://localhost:8080/meetings/" + meeting.getId();
         if(session.getAttribute("user") != null) {
-            Participant user = (Participant) session.getAttribute("user");
-            meeting.addParticipant(user);
-            user.setMeeting_id(meeting.getId());
-            participantRepository.save(user);
+            User user = (User) session.getAttribute("user");
+            meeting.addUser(user);
+            userRepository.save(user);
```
Comment (cjh5414) [inline src/main/java/us/controller/MeetingController.java]: `user` have to save `meeting`.

File: src/main/java/us/model/User.java [hunk 3]
```diff
@@ -1,45 +1,39 @@
...
+import java.util.List;
 
 @Entity
-public class Participant {
+public class User {
 
     @Id
     @Column(name="id")
-    @GeneratedValue(strategy=GenerationType.AUTO)
+    //@GeneratedValue(strategy=GenerationType.AUTO)
```
Comment (cjh5414) [inline src/main/java/us/model/User.java]: remove unnecessary this comment

File: src/main/java/us/controller/MeetingController.java [hunk 4]
```diff
@@ -20,12 +22,14 @@
     @Autowired
     private MeetingRepository meetingRepository;
     @Autowired
-    private ParticipantRepository participantRepository;
+    private UserRepository userRepository;
+    @Autowired
+    private ExpenseRepository expenseRepository;
```
Comment (NESOY) [inline src/main/java/us/controller/MeetingController.java]: This Field is not used 
Plz Remove it

File: src/main/java/us/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package us.repository;
+
+import org.springframework.data.repository.CrudRepository;
+import us.model.User;
+
+import javax.transaction.Transactional;
+
+@Transactional
```
Comment (cjh5414) [inline src/main/java/us/repository/UserRepository.java]: Remove `@Transcational` annotation.

File: src/main/java/us/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+public interface UserRepository extends CrudRepository<User, Integer> {
+    User findById(int id);
+}
```
Comment (cjh5414) [inline src/main/java/us/repository/UserRepository.java]: `findById` can be replaced `findOne`

**Review by cjh5414 — CHANGES_REQUESTED** [review]

**Review by NESOY — CHANGES_REQUESTED** [review]

**Review by cjh5414 — APPROVED** [review]

**Review by NESOY — APPROVED** [review]

**Pull request merged**